public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts

```diff
@@ -71,12 +70,15 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
     });
   }
 
-  async getWorkspaces(subscription: string): Promise<AzureLogsVariable[]> {
+  async getWorkspaces(subscription: string, returnCustomerID?: boolean): Promise<AzureLogsVariable[]> {
```

Comment:
I don't think we need this `returnCustomerID`. I think getWorkspaces should always return resource URIs

It looks like the only place the 'customerID' is being used is in the testDatasource method to fetch metadata for the resource, which should also work with the resource URI.

Load a LA workspace into a logs query and use the network inspector to observe the request to get metadata work with the resource URI. That should allow us to get rid of the isGUIDish tenery - it'll never be a guid.

Reply:
the request was failing just because a double slash :facepalm: